webdriver/tests/bidi/browsing_context/navigate/navigate.py

```diff
@@ -108,3 +109,44 @@ async def test_same_document_navigation_in_before_unload(bidi_session, new_tab,
 
     url_after = url_before.replace("empty.html", "other.html")
     await navigate_and_assert(bidi_session, new_tab, url_after, "complete")
+
+
+@pytest.mark.parametrize(
+    "script",
+    [
+        "<script>window.location='{url}'</script>",
+        """<script>window.addEventListener('DOMContentLoaded', () => {{
+            window.location = '{url}';
+        }});</script>""",
+        """<script>window.addEventListener('load', () => {{
+            window.location = '{url}';
+       }});</script>""",
+    ],
+    ids=["Redirect immediately", "Redirect on DOMContentLoaded", "Redirect on load"],
+)
+@pytest.mark.parametrize("wait", ["none", "interactive", "complete"])
+async def test_interrupted_navigation(
```

Comment:
we should also probably assert that the redirect actually happens by waiting for the redirect navigation events

Reply:
Sounds good will add.

Reply:
As discussed in the other thread, I will wait for the final browsingContext.load event.